osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerMatchSettingsOverlay.cs

```diff
@@ -510,28 +492,34 @@ private void onSuccess(Room room) => Schedule(() =>
                 applyingSettingsOperation = null;
             });
 
-            private void onError(string text) => Schedule(() =>
+            private void onError(Exception? exception, string description)
             {
-                Debug.Assert(applyingSettingsOperation != null);
+                if (exception is AggregateException aggregateException)
+                    exception = aggregateException.AsSingular();
 
-                // see https://github.com/ppy/osu-web/blob/2c97aaeb64fb4ed97c747d8383a35b30f57428c7/app/Models/Multiplayer/PlaylistItem.php#L48.
-                const string not_found_prefix = "beatmaps not found:";
+                string message = exception?.GetHubExceptionMessage() ?? $"{description} ({exception?.Message})";
 
-                if (text.StartsWith(not_found_prefix, StringComparison.Ordinal))
-                {
-                    ErrorText.Text = "The selected beatmap is not available online.";
-                    SelectedItem.Value?.MarkInvalid();
-                }
-                else
+                Schedule(() =>
                 {
-                    ErrorText.Text = text;
-                }
+                    Debug.Assert(applyingSettingsOperation != null);
 
-                ErrorText.FadeIn(50);
+                    // see https://github.com/ppy/osu-web/blob/2c97aaeb64fb4ed97c747d8383a35b30f57428c7/app/Models/Multiplayer/PlaylistItem.php#L48.
+                    const string not_found_prefix = "beatmaps not found:";
 
-                applyingSettingsOperation.Dispose();
-                applyingSettingsOperation = null;
-            });
+                    if (message.StartsWith(not_found_prefix, StringComparison.Ordinal))
+                    {
+                        ErrorText.Text = "The selected beatmap is not available online.";
+                        room.Playlist.SingleOrDefault()?.MarkInvalid();
```

Comment:
Slight concern: This change bothers me because it assumes that `room.Playlist.SingleOrDefault()` is going to be the same object reference as the UI-facing, displayed, playlist item, and that feels like a brittle assumption.

Reply:
In the case of multiplayer, this is always the case for now. Unsure if this going to change or if it's going to go in the opposite direction - rooms being able to be started with _no_ items in the playlist (matchmaking will eventually allow that, so it's not a stretch and would be a UX improvement here).

https://github.com/ppy/osu/blob/66b8b527e3e5b57a6529941f49a18ee8193c1a07/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSongSelect.cs#L114-L118